application/api/example.1/add.js

```diff
@@ -5,9 +5,40 @@
   },
 
   method: async ({ a, b }) => {
+    new api.example.add.CustomError('EARGA');
+    if (typeof a !== 'number') return new DomainError('EARGA');
+    if (typeof b !== 'number') {
+      return new api.example.example.CustomError('EARGB');
+    }
+    if (Number.isNaN(a)) throw Error('Not a number: a');
+    if (Number.isNaN(b)) throw Error('Not a number: b');
     const result = a + b;
     return result;
   },
 
   returns: 'number',
+
+  errors: {
+    EARGA: 'Invalid argument: a',
+    EARGB: 'Invalid argument: b',
+  },
+
+  onError(error) {
+    if (error.code in this.errors) {
```

Comment:
It is unlikely that you will know all errors that are thrown from inside the domain. Or more accurately - unlikely that you want to list them in every method. It adds a lot of chore to the api definition.
IMO it is better to manage this per-error and let domain decide which to throw.